src/uiHelpers_nanox.c

```diff
@@ -0,0 +1,113 @@
+#include <os_io_seproxyhal.h>
+#if defined(TARGET_NANOX)
+#include <string.h>
+
+#include "uiHelpers.h"
+#include "ux.h"
+#include "assert.h"
+#include "io.h"
+#include "utils.h"
+#include "securityPolicy.h"
+
+// Helper macro for better astyle formatting of UX_FLOW definitions
+#define LINES(...) { __VA_ARGS__ }
+
+void scrolling_confirm()
+{
+    // TODO(ppershing): exception handling
+    TRY_CATCH_UI({
+        scrollingState_t* ctx = scrollingState;
+        uiCallback_confirm(&ctx->callback);
+    });
+}
+
+UX_FLOW_DEF_VALID(
+    ux_display_scrolling_flow_1_step,
+    bnnn_paging,
+    scrolling_confirm(),
+    LINES(
+        (char *) &displayState.scrolling.header,
+        (char *) &displayState.scrolling.fullText
+    )
+);
+
+const ux_flow_step_t * const ux_scrolling_flow [] = {
+    &ux_display_scrolling_flow_1_step,
+    FLOW_END_STEP,
+};
+
+UX_FLOW_DEF_NOCB(
```

Comment:
Can we move busy to the top (or bottom)? It feels unnatural to have busy flow (which is fundamentally different) inbetween the two others. Also, maybe separating flows with `// Flow xyz` comment would be nice